docs/release.md

```diff
@@ -0,0 +1,73 @@
+# Releasing
+
+[semver]: https://semver.org
+[`VERSION`]: ../VERSION
+[`VERSION-DEV`]: ../VERSION-DEV
+
+This document describes how to perform a [semver] release.
+
+## Release artifacts
+1. Branch `release-vMajor.Minor` for each `Major.Minor.*` version. Example `release-v0.8`
+2. Tag `vMajor.Minor.Patch` for each `Major.Minor.Patch` version. Example `v0.8.1`
+3. All in one deployment yaml for the Application controller
+4. Container image for the Application controller
+
+## Release roles
+
+Only Repo Owners can create branches in the [Application Repo](https://github.com/kubernetes-sigs/application). Developers who fork the repo can create releases in their repo as well.
+
+#### Version files
+For official releases [`VERSION`] file is used.
+For developers [`VERSION-DEV`] file is used.
+
+The default file used is [`VERSION-DEV`].
+To use [`VERSION`], set the `VERSION_FILE` env variable.
+
+Developers should edit the [`VERSION-DEV`] file to set their choice of container registry and version.
+
+## Release procedure
```

Comment:
It's not clear to me which task I should run in order to make a release.
Should I run `make release-branch` first, and then `make release-tag`.
- If so, `release-tag` depends on `release-branch`, run `make release-tag` alone should be okay.
- If not, then `make release-branch` alone doesn't generate any release tag. What do we do with a dangled release branch?

Reply:
Release branch is usually created once per release $major.$minor where as release tag is major.minor.patch 
So for the first time release branch would be created. 

Calling release-branch explicitly is a good practice.